### DISCUSSION TITLE
Help with inital code

### DISCUSSION THREAD

**Discussion by Binocularbath** [Serene-Arc/bulk-downloader-for-reddit#660] [Q&A]
Hello everyone

After spending years looking at github projects that I would like to use but not being able to because I didn't understand the way it works, I've decided to give it a go. This project looks awesome and its exactly what I'm looking for. My thanks to the creators.

I have downloaded python. Downloaded the code for this project then started up cmd.

Heres what I've done so far and the challenges I've come across.

First I ran:

> **py -m pip install bdfr**

It said requirements already satisfied

Then:

> **py -m bdfr archive**

I got: 

> **Usage: python -m bdfr archive [OPTIONS] DIRECTORY
> Try 'python -m bdfr archive --help' for help.**
> 
> **Error: Missing argument 'DIRECTORY'.**
> 
> **C:\Github Projects\bulk-downloader-for-reddit-development**

I also tried:

**> py -m bdfr download C:\Github Projects\redditsave downloads –user MYREDDITUSERNAME –submitted -L 100**

and got:

> 
> **Usage: python -m bdfr download [OPTIONS] DIRECTORY
> Try 'python -m bdfr download --help' for help.**
> 
> **Error: Got unexpected extra arguments (Projects\redditsave downloads MYREDDITUSERNAME )**

As I mention, this is my first time trying anything like this and I'm really our of my depth. Any guidance would be awesome. Id love to learn some new skills, be able to use other projects and who knows, maybe even contribute to other projects in the future.

Any advice majorly appreciated

**Comment by Binocularbath** [comment]
UPDATE: Ive been told i cant have spaces in the path so I added a dash to the folder names to rectify this but then got:

> **2022-09-13 22:25:35,049 - bdfr.connector - ERROR] - Source ubmitted does not exist
> [2022-09-13 22:25:35,051 - root - INFO] - Program complete**

**Reply by Serene-Arc** [reply]
You need two dashes in front of submitted. It's recognising only the letter `-s` and then reading the next thing as a source, which is the string `ubmitted`. Do `--submitted` and the error will disappear.

**Reply by Binocularbath** [reply]
worked perfectly, thank you. Not sure why I had that. Sorry, this is the first time I've ever used python

**Reply by Serene-Arc** [reply]
all good. all of the options are like that; you can see the single letter shortcut that some of them have, and then a longer string that's called by two dashes.
